packages/synapse-react-client/src/synapse-queries/subscription/useSubscription.ts

```diff
@@ -57,6 +63,70 @@ export function useGetSubscription(
   )
 }
 
+export function useGetAllSubscriptions(
+  query: SubscriptionQuery,
+  options?: UseInfiniteQueryOptions<
+    SubscriptionPagedResults,
+    SynapseClientError,
+    Subscription
+  >,
+  queryKeyOverride?: QueryKey,
+) {
+  const { accessToken, keyFactory } = useSynapseContext()
+  const queryClient = useQueryClient()
+
+  return useInfiniteQuery<
+    SubscriptionPagedResults,
+    SynapseClientError,
+    Subscription
+  >(
+    queryKeyOverride ?? keyFactory.getAllSubscriptionsQueryKey(query),
+    async context => {
+      const offset = context.pageParam as number | undefined
+      return await SynapseClient.getAllSubscriptions(
+        accessToken,
+        10,
+        offset,
+        query,
+      )
+    },
+    {
+      ...options,
+      onSuccess: data => {
+        // Set the query data for each individual subscription that was retrieved
+        data.pages.forEach(subscription => {
+          queryClient.setQueryData(
+            keyFactory.getSubscriptionQueryKey(
+              subscription.objectId,
+              subscription.objectType,
+            ),
+            subscription,
+          )
+        })
```

Comment:
Pre-filling the cache here with each list item prevents unnecessary requests when each individual subscription topic component is rendered and calls its own `useGetSubscription`.